db/system_keyspace.cc

```diff
@@ -2118,6 +2120,217 @@ class protocol_servers_table : public memtable_filling_virtual_table {
     }
 };
 
+class runtime_info_table : public memtable_filling_virtual_table {
+private:
+    service::storage_service& _ss;
```

Comment:
This is sad, storage_service is a high-level thing and system_keyspace is a low-level thing (and used by storage_service too). We should register the table outside system_keyspace, maybe @xemul already has some helpers for it.

Reply:
Yes, and also these virtual tables started an include creep in `db/system_keyspace.cc`.
I like the idea of a virtual table service, which is passed down to interested parties who can then register their virtual tables.

Reply:
This hackathon I had a branch with gossiper and cql_transport calling smth like `db:system_keyspace::register_virtual_table(my_vtable(*this))` on their start (and similar in main to register vtable for db::config). It's not exactly a virtual table service, but rather a re-use of system_keyspace in a layer-correct way. However I didn't make it really work until hackathon finished :( so nothing to send as RFC yet, but that's the way I plan to proceed.